src/common/AppConfig.ts

```diff
@@ -0,0 +1,72 @@
+import { DEFAULT_SETTINGS, Settings } from "../types/App.types";
+
+/**
+ * A singleton class to manage global properties.
+ * 
+ * If app settings aren't provided during initialisation, default settings are applied.
+ */
+export class AppConfig {
+    private static instance: AppConfig;
+    private config: { [key: string]: any; };
+
+    /**
+     * Private constructor to avoid multiple instances.
+     * 
+     * @param {object} [_config] - Configuration object
+     */
+    private constructor(_config: { [key: string]: any }) {
+        console.log("AppConfig().constructor() || _config", _config);
+
+        this.config = { ..._config };
+    }
+
+    /**
+     * Public method to initialise AppConfig.
+     * 
+     * If app settings aren't provided during initialisation, default settings are injected to the configuration.
+     * If it's already been initialised, the new properties are injected to the configuration. The existing properties are updated as well.
+     * 
+     * @param {object} [_config] - Optional configuration
+     * @returns The instance of AppConfig
+     */
+    public static getInstance(_config?: { [key: string]: any }): AppConfig {
+        console.log("AppConfig().getInstance()");
+
+        if (!AppConfig.instance) {
+            // inject the settings if not provided during initialisation
+            if (!_config || !_config.settings) {
+                _config = { ...{ settings: DEFAULT_SETTINGS } };
+            }
```

Comment:
after discussing with @syapar, it's agreed to update the logic _-to avoid using index signature,_ alongside the config structure to align the requirements as well as their default values.
https://github.com/irgat/soloTest/blob/850758dbc02694d5a3ffd3992384460258d5a57a/src/common/AppConfig.ts#L10

Reply:
related commit: https://github.com/irgat/soloTest/pull/1/commits/5af5afe81633ab8eb888aa79d4fb87f6f75cc197

---

the new logic decouples the manifest and app settings, default values are applied if they aren't provided.
https://github.com/irgat/soloTest/blob/f5b1589ff3b5a1a0f980e6e209ba6795c07ccf66/src/common/AppConfig.ts#L25-L44